Add an utility fn to strip http(s) out of an URL

lamosty opened the pull request:
I think it's a good addition to Calypso as I've seen .replace( /^https?:\/\//, '' ) on too many places.

cc @gwwar 

Test live: https://calypso.live/?branch=add/without-http-utility-fn

gwwar commented on client/lib/site/utils.js:
```diff
@@ -172,5 +172,18 @@ export default {
+			return null;
+		}
+
+		return url.replace( /^https?:\/\//, '' );
```
If our regexes are static, I prefer defining them outside of the calling function so they don't need to be created each time.

gwwar commented:
This helper method feels like it doesn't belong on the site util and instead on a url util. Could you also update the PR with a few examples you wanted to clean up.

ockham commented:
> This helper method feels like it doesn't belong on the site util and instead on a url util. 

Yep, like `lib/url`.

ockham commented:
Also, for best practice -- maybe include a test? :-D

lamosty commented:
Thanks for the reviews! I moved the fn into `lib/url`, used a Regex literal for better perf, replaced **all** occurrences of `/^https?:\/\//` with the fn (manually) and finally, added a few tests.

Could you give it another look, please?

gwwar commented:
Looking at the example tests, are we using this to get the host? For example should the following `http://myname.com?foo=bar#anchor` return `myname.com` or `myname.com?foo=bar#anchor`?

If it's the first, we can do:
```javascript
import { parse } from 'url';
const host = parse( myExampleUrl ).host;
```

lamosty commented:
> Looking at the example tests, are we using this to get the host? 

Uhh, I overlooked the fact that we have the `url` lib in codebase. My motivation for creating this function was to replace all the occurrences of `.replace( /^https?:\/\//, '' )` with it. I think that it doesn't return the host, it simply removes the initial `http(s)://` from an URL. In many cases, that produces a host. But I think it should also return `myname.com?foo=bar#anchor` if you pass `http://myname.com?foo=bar#anchor` to it.

I'm going to add your example to the tests so it's more clear what it does.

gwwar commented on client/lib/url/index.js:
```diff
@@ -26,8 +26,24 @@ function isHttps( url ) {
+ * @return {?String}     URL without the initial http(s)
+ */
+function withoutHttp( url ) {
+	if ( ! url ) {
```
Empty strings will be falsey. It may be safer to return `url` in this case if you look at where this is being called.

lamosty replied:
Oh, agreed. Great catch. Will fix!

gwwar commented:
👍 Code here looks good. Feel free to 🚢 when you're ready.

lamosty commented:
Fixed the empty string issue. Going to merge. Thanks for the review!
